homeassistant/components/elmax/config_flow.py

```diff
@@ -78,55 +74,54 @@ async def async_step_user(
             client = Elmax(username=username, password=password)
             await client.login()
 
-            # If the login succeeded, retrieve the list of available panels and filter the online ones
-            online_panels = [x for x in await client.list_control_panels() if x.online]
-
-            # If no online panel was found, we display an error in the next UI.
-            panels = list(online_panels)
-            if len(panels) < 1:
-                raise NoOnlinePanelsError()
-
-            # Show the panel selection.
-            # We want the user to choose the panel using the associated name, we set up a mapping
-            # dictionary to handle that case.
-            panel_names: dict[str, str] = {}
-            username = client.get_authenticated_username()
-            for panel in panels:
-                _store_panel_by_name(
-                    panel=panel, username=username, panel_names=panel_names
-                )
-
-            self._client = client
-            self._panel_names = panel_names
-            schema = vol.Schema(
-                {
-                    vol.Required(CONF_ELMAX_PANEL_NAME): vol.In(
-                        self._panel_names.keys()
-                    ),
-                    vol.Required(CONF_ELMAX_PANEL_PIN, default="000000"): str,
-                }
-            )
-            self._panels_schema = schema
-            self._username = username
-            self._password = password
+        except ElmaxBadLoginError:
             return self.async_show_form(
-                step_id="panels", data_schema=schema, errors=errors
+                step_id="user",
+                data_schema=LOGIN_FORM_SCHEMA,
+                errors={"base": "invalid_auth"},
             )
-
-        except ElmaxBadLoginError:
-            _LOGGER.error("Wrong credentials or failed login")
-            errors["base"] = "bad_auth"
-        except NoOnlinePanelsError:
-            _LOGGER.warning("No online device panel was found")
-            errors["base"] = "no_panel_online"
         except ElmaxNetworkError:
             _LOGGER.exception("A network error occurred")
-            errors["base"] = "network_error"
+            return self.async_show_form(
+                step_id="user",
+                data_schema=LOGIN_FORM_SCHEMA,
+                errors={"base": "network_error"},
```

Comment:
If the network error means we can't connect we usually use the standard `cannot_connect` error string.

Reply:
The the [low-level library](https://github.com/albertogeniola/elmax-api/blob/0ca9438c28e11d56ebc843fee1da81add3979908/elmax_api/http.py#L157) catches both connection issues and read-timeouts. Usually this falls into the "cannot-connect" case but also covers the case when the server sends a partial response (i.e. it sends headers but not the body).

```python
# Wrap any other HTTP/NETWORK error
except (httpx.ConnectError, httpx.ReadTimeout):
    _LOGGER.exception("An unhandled error occurred while executing API Call.")
    raise ElmaxNetworkError("A network error occurred")
```

I think we can leave this as it is now.